src/include/migraphx/compile_src.hpp

```diff
@@ -37,8 +37,18 @@ inline namespace MIGRAPHX_INLINE_NS {
 struct src_file
 {
     fs::path path;
-    std::pair<const char*, const char*> content;
-    std::size_t len() const { return content.second - content.first; }
+    std::string_view content;
+
+    src_file() = delete;
```

Comment:
Dont delete the default constructor.

Reply:
My thoughts are the structure `src_file` is implementing the association pattern. Although the default constructor does not harm, it does not help to use the structure optimally. One may first construct an instance of the structure and then assign values to its members to associate the file name with its content. Optimally is to call the constructor with arguments and let the compiler optimize things for us. In the case of containers, deleting the default constructor forces us to use in-place construction, which is why optimal than calling, for instance, e.g., `push_back` from `std::vector` (however, that may not be the best example, but I guess you know what I mean).
I will remove that change if you insist on keeping the default constructor.

Reply:
I would prefer to keep the type regular. This also allows improving readability by using the member variables to construct.